tests/test_pytest_mock.py

```diff
@@ -185,23 +194,35 @@ def test_mocker_resetall(mocker: MockerFixture) -> None:
     listdir = mocker.patch("os.listdir", return_value="foo")
     open = mocker.patch("os.open", side_effect=["bar", "baz"])
 
+    mocked_object = mocker.create_autospec(TestObject)
+    mocked_object.run.return_value = "mocked"
+
     assert listdir("/tmp") == "foo"
     assert open("/tmp/foo.txt") == "bar"
+    assert mocked_object.run() == "mocked"
     listdir.assert_called_once_with("/tmp")
     open.assert_called_once_with("/tmp/foo.txt")
+    mocked_object.run.assert_called_once()
 
     mocker.resetall()
 
     assert not listdir.called
     assert not open.called
+    assert not mocked_object.called
     assert listdir.return_value == "foo"
     assert list(open.side_effect) == ["baz"]
+    assert mocked_object.run.return_value == "mocked"
 
     mocker.resetall(return_value=True, side_effect=True)
 
     assert isinstance(listdir.return_value, mocker.Mock)
     assert open.side_effect is None
 
+    if sys.version_info >= (3, 9):
+        # The reset on child mocks have been implemented in 3.9
+        # https://bugs.python.org/issue38932
+        assert mocked_object.run.return_value != "mocked"
```

Comment:
Don't really like this check here, maybe test needs to be split, and then it's easier to use `pytest.mark.skipif` to skip for 3.8 and below.

Reply:
Or maybe have a completely separate test for resetting child mocks

Reply:
The check is fine, when we drop support for Python versions it is common to search for `sys.version_info` checks like these. 👍